testsuite/test-runner/src/main/java/org/wildfly/core/testrunner/Server.java

```diff
@@ -59,7 +61,7 @@
     private Thread shutdownThread;
 
     private volatile Process process;
-    private volatile ManagementClient client;
+    private final ManagementClient client = createClient();
```

Comment:
I almost wonder if this should just be `new ManagementClient(new DelegatingModelControllerClient(ServerClientProvider.INSTANCE), managementAddress, managementPort, managementProtocol)` so initially no `ModelControllerClient` is created. Essentially only leaving the creation of the MCC to `start()` and `reload()` operations.